act/registry.go

```diff
@@ -402,6 +405,53 @@ func runActionWithTimeout(
 	}
 }
 
+// RunAll run all the actions in the outputs and returns the end result.
+func (r *Registry) RunAll(result map[string]any) map[string]any {
+	if _, exists := result[sdkAct.Outputs]; !exists {
+		r.Logger.Debug().Msg("Outputs key is not present, returning the result as-is")
+		return result
+	}
+
+	var (
+		outputs []*sdkAct.Output
+		ok      bool
+	)
+	if outputs, ok = result[sdkAct.Outputs].([]*sdkAct.Output); !ok || len(outputs) == 0 {
+		// If the outputs are nil or empty, we should delete the key from the result.
+		delete(result, sdkAct.Outputs)
+		r.Logger.Debug().Msg("Outputs are nil or empty, returning the result as-is")
+		return result
+	}
+
+	endResult := make(map[string]any)
+	for _, output := range outputs {
+		if !cast.ToBool(output.Verdict) {
+			r.Logger.Debug().Msg(
+				"Skipping the action, because the verdict of the policy execution is false")
+			continue
+		}
+		runResult, err := r.Run(output, WithResult(result), WithLogger(r.Logger))
+		// If the action is async and we received a sentinel error, don't log the error.
+		if err != nil && !errors.Is(err, gerr.ErrAsyncAction) {
+			r.Logger.Error().Err(err).Msg("Error running policy")
+		}
+		// Each action should return a map.
+		if v, ok := runResult.(map[string]any); ok {
+			endResult = v
+		}
+	}
+	return endResult
+}
+
+// ShouldTerminate checks if any of the actions are terminal, indicating that the request
+// should be terminated.
+// This is an optimization to avoid executing the actions' functions unnecessarily.
+// The __terminal__ field is only set when an action intends to terminate the request.
+func (r *Registry) ShouldTerminate(result map[string]any) bool {
+	_, ok := result[sdkAct.Terminal]
+	return ok && cast.ToBool(result[sdkAct.Terminal])
```

Comment:
Nit:
Avoids casting if the key doesn’t exist
Improved type safety with the use of ok checks
```suggestion
	terminalVal, exists := result[sdkAct.Terminal]
	if !exists {
		r.Logger.Debug().Msg("'__Terminal__' key not found, request will continue.")
		return false
	}

	shouldTerminate, ok := cast.TryBool(terminalVal)
	if !ok {
		r.Logger.Warn().Msg("'__Terminal__' key exists but cannot be cast to a boolean.")
		return false
	}

	if shouldTerminate {
		r.Logger.Info().Msg("Request is marked as terminal. Terminating.")
	}
	return shouldTerminate
```